src/providers/dart_rename_provider.ts

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+import { window, workspace, RenameProvider, WorkspaceEdit, TextDocument, Position, CancellationToken, Uri, TextEdit, Range } from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+import * as as from "../analysis/analysis_server_types";
+
+export class DartRenameProvider implements RenameProvider {	
+	private analyzer: Analyzer;
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+	
+	provideRenameEdits(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {		
+		return workspace.saveAll(false).then(() => {
```

Comment:
Do we actually need to `saveAll` first? Seems unexpected, and the analysis service has all of the in-memory edits. If I was just working in a new file and renaming stuff, I'm not sure I'd expect that (obviously if it's required for some reason, it's not a huge issue!).